torchaudio/_internal/fft.py

```diff
@@ -0,0 +1,27 @@
+"""Compatibility module for fft-related functions
+
+In PyTorch 1.7, the new `torch.fft` module was introduced.
+
+To use this new module, one has to explicitly import `torch.fft`. however this will change
+the reference `torch.fft` is pointing from function to module.
+And this change takes effect not only in the client code but also in already-imported libraries too.
+Similarly, if a library does the explicit import, the rest of the application code must use the
+`torch.fft.fft` function.
+
+For this reason, to migrate the deprecated functions of fft-family, we need to use the new
+implementation under `torch.fft` without explicitly importing `torch.fft` module.
+
+This module provides a simple interface for the migration, abstracting away
+the access to the underlying C functions.
+
+Once the deprecated functions are removed from PyTorch and `torch.fft` starts to always represent
+the new module, we can get rid of this module and call functions under `torch.fft` directly.
+"""
+from typing import Optional
+
+import torch
+
+
+def rfft(input: torch.Tensor, n: Optional[int] = None, dim: int = -1, norm: Optional[str] = None) -> torch.Tensor:
+    # see: https://pytorch.org/docs/master/fft.html#torch.fft.rfft
+    return torch._C._fft.fft_rfft(input, n, dim, norm)
```

Comment:
I wrote a response in the document you've shared about why this mapping is more complicated. I can probably write a mapping from `torch.rfft` to `torch.fft.rfft` for you, though. Let me know.

Reply:
or you could use `torch.view_as_real` at the end, although I agree with @mruberry that it would be kind of wasteful to use the `torch.fft.rfft` before we migrate audio to start using complex tensors.

Reply:
@mruberry, @anjali411 

> I wrote a response in the document you've shared about why this mapping is more complicated. 

I looked at your response and it was very helpful, however that document was for general cases, and here we are dealing with the migration of specific use cases, and I think the concerns brought up are irrelevant. Here are the reasons;

The all `torch.rfft` usages in `torchaudio` are
1. Inputs are always 2D and `onesided=True`.
    This means that we can simply migrate to `torch.fft.rfft`, and it is sufficient. We do no need to consider `torch.fft.rffftn`
2. No normalization is required.
    We can pass `norm=None` to `torch.fft.rfft`, which is default so in my code change it is omitted.
3. The complex values are immediately used to compute power.
    Look at the change in `functional.py` and `kaldi.py`. `rfft` is used as a mean of computing the power of the input signal in frequency domain. Therefore, with the appropriate changes (like using `.abs` on complex dtype), no computation is wasted, or no unnecessary computation is introduced.

> I can probably write a mapping from `torch.rfft` to `torch.fft.rfft`

For torchaudio's usecase, the general mapping from `torch.rfft` to `torch.fft.rfft` is not necessary.

Reply:
Note that this compatibility module is not for migration from `torch.rfft` to `torch.fft.rfft` but for TorchScript compatibility `torch.fft.rfft` (and others coming up) while avoiding explicit import of `torch.fft`.

Reply:
If there is a function that computes power spectrum (which is a real value) directly, then we do not need to use `torch.fft.rfft` but otherwise, the use of complex dtype here as an intermediate expression makes sense and it is not wasteful.

Reply:
Interesting. Thanks for the additional context, @mthrok. Would it make sense, then, to limit the signature of this function or add a comment explaining that it only does a limited translation from torch.rfft to torch.fft.rfft?

Also, I'm not sure how many versions of PyTorch torchaudio supports, but torch.fft.rfft will only be available in PyTorch 1.7+, so previous versions of PyTorch will still need to call torch.rfft.

Reply:
@mruberry 

> Would it make sense, then, to limit the signature of this function or add a comment explaining that it only does a limited translation from torch.rfft to torch.fft.rfft?

I gave some thoughts on it, and we could do that but I think having the same signature as `torch.fft.rfft` has advantages when it comes to maintainability.
1. If someone wants to add a new functionality that uses `rfft`, he/she can simply use the same signature as `torch.fft.rfft`, that will free him/her from having to consider the relationship between our abstraction function and other functions of torchaudio that use this abstraction function. (in short, the person working on the new feature can use this abstraction as a drop-in replacement)
2. similar to 1., in future when we PyTorch is done with the immigration, we can simply replace the abstraction function path with the actual PyTorch implementation. (`torchaudio._internal.fft.rfft` -> `torch.fft.rfft`)

> Also, I'm not sure how many versions of PyTorch torchaudio supports, but torch.fft.rfft will only be available in PyTorch 1.7+, so previous versions of PyTorch will still need to call torch.rfft.

That is okay for domain libraries. We clearly state that domain libraries are expected/tested to work with the version of PyTorch that is released at the same time. So all the work on master branch expects master version (or the next latest stable release) of PyTorch.